sympy/solvers/diophantine.py

```diff
@@ -1455,6 +1456,8 @@ def cornacchia(a, b, m):
             s, _exact = integer_nthroot(m1, 2)
             if _exact:
                 if a == b and r > s:
+                    pass
```

Comment:
It seems that this line and the one above could be omitted without changing the output.